collectd/files/plugin/openstack_nova.py

```diff
@@ -52,18 +52,30 @@ def collect(self):
                                  'service': service,
                                  'state': state})
 
-        for service in aggregated_workers:
-            totalw = sum(aggregated_workers[service].values())
+        for service in (set(aggregated_workers.keys()) | set(
+                ('compute', 'scheduler', 'conductor', 'cert', 'consoleauth'))):
```

Comment:
I think you can make it more elegant with the `union` function, because you don't need to create a set for the `('compute', …)` tuple:

```py
    for service in set(aggregated.workers.keys()).union(('compute', …):
```